### DISCUSSION TITLE
Merging SQL Connection Factories

### DISCUSSION THREAD

**Discussion by preardon** [BrighterCommand/Brighter#1518] [Ideas]
At the minute there is a IMsSqlInboxConnectionFactory and a IMsSqlOutboxConnectionFactory, as well as soon a IMsSqlMessagingGatewayConnectionFactory.

I am thinking that we could make a shared project to hold a IMsSqlDbConnectionConnectionFactory that would replace the three above and also hold the SQL Authentication Connection Factory.

One Question I do have is if we should have is should we also create another Project for the AzureSQL Connection Factories so that you don't have to Pull Azure.Identity in as a Dependency if you do not need it.

What are your thoughts @iancooper I can get this done as Part of the MsSQL Gateway Tests

**Comment by iancooper** [comment]
Hi @preardon,

Not sure if you remember, I discussed this under one of your PRs. The only downside I can see is increased dependencies. It might also be worth thinking about the option of a Box assembly, that has both the Inbox and the Outbox in it. A lot of services both send a receive, so it might not be the dependency on unneeded code we worry about, a smaller number of projects would help.

It would mean folks have to change their packages,  but I suspect that it is better to do that at a major release, so otherwise we would put it off until v10.

To be clear, I think if we combine one inbox/outbox, then we combine them all: so Paramore.Brighter.Box.MSSQL, Paramore.Brighter.Box.Postgres etc by Db

@holytshirt What are your thoughts? Simplification with a few cases where folks who don't need both get them?

**Comment by preardon** [comment]
@iancooper definitely remember the discussion on the PR, that's why I raised this discussion topic to make sure we're on the same page before I start the work (I'll add Azure while doing the integration tests)

Even if we combine inbox and outbox it we would still need SQLDBConnection Factory across box and Messaging Gateway. I don't really love having a lot of small projects as they are extra dependencies. I feel for now perhaps I Just duplicate the code for now and we can look at combining for v10

**Comment by preardon** [comment]
That being said I'm happy to do the box work in time for v9 if you wanted

**Reply by iancooper** [reply]
It may be the point to call v9, fix and ship, then talk about 9.x and 10

**Reply by preardon** [reply]
Yup completely agree, I am keen to get v9 out so I can Start using it in anger at work, I've got a piece of work starting next week that I would love to use it for

**Comment by preardon** [comment]
This has been fixed in v9
